crates/frost/src/signer.rs

```diff
@@ -120,44 +122,52 @@ impl FrostSigner {
 
     pub async fn sign_round_1(&self, request: SignRound1Request) -> Result<SignRound1Response, SignerError> {
         let user_id = request.user_id;
+        let session_id = request.session_id.clone();
         let tweak = request.tweak;
+
         let state = self.user_storage.get_user_state(user_id.clone()).await?;
 
         match state {
-            Some(SignerUserState::DkgFinalized { key_package }) => {
+            Some(SignerUserState::DkgFinalized { key_package}) => {
                 let tweak_key_package = match tweak.clone() {
                     Some(tweak) => key_package.clone().tweak(Some(tweak.to_vec())),
                     None => key_package.clone(),
                 };
                 let (nonces, commitments) =
                     frost_secp256k1_tr::round1::commit(tweak_key_package.signing_share(), &mut OsRng);
 
-                self.user_storage
-                    .set_user_state(
+                self.session_storage
+                    .set_session_state(
                         user_id.clone(),
-                        SignerUserState::SigningRound1 {
-                            key_package: key_package,
-                            tweak: tweak,
+                        session_id.clone(),
+                        SignerSessionState::SigningRound1 {
+                            key_package,
+                            tweak,
                             nonces,
                         },
                     )
                     .await?;
-                Ok(SignRound1Response { user_id, commitments })
+                Ok(SignRound1Response { user_id, session_id, commitments })
             }
             _ => {
-                return Err(SignerError::InvalidUserState(
+                Err(SignerError::InvalidUserState(
                     "User state is not SigningRound1".to_string(),
-                ));
+                ))
             }
         }
     }
 
     pub async fn sign_round_2(&self, request: SignRound2Request) -> Result<SignRound2Response, SignerError> {
-        let user_id = request.user_id;
-        let state = self.user_storage.get_user_state(user_id.clone()).await?;
+        let user_id = request.user_id.clone();
+        let session_id = request.session_id.clone();
```

Comment:
```suggestion
        let SignRound2Request{ user_id, session_id,signing_package: round2_sesssion_package} = request;
```
and use in further code `round2_sesssion_package` variable where you're using `request.signing_package`